### PR TITLE
Fix TestPriorityLevelProperty test

### PR DESCRIPTION
**Pull request by stephentoub**
If the target thread goes away between the time we grab it and try to manipulate it, the test fails.  Instead, just create a thread in the current process, as we can then target that one directly and ensure it's always in a good state for testing against.

cc: @wtgodbe, @krwq, @ViktorHofer

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Is there a failed run you can link to?

**Comment by stephentoub** [comment]
https://dev.azure.com/dnceng/public/_build/results?buildId=176695&view=ms.vss-test-web.build-test-results-tab&runId=4248718&resultId=280124&paneView=debug
```

Error message
System.InvalidOperationException : The request cannot be processed because the thread (3620) has exited.

Stack trace
   at System.Diagnostics.ProcessManager.OpenThread(Int32 threadId, Int32 access) in D:\a\1\s\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessManager.Windows.cs:line 219
   at System.Diagnostics.ProcessThread.OpenThreadHandle(Int32 access) in D:\a\1\s\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessThread.Windows.cs:line 175
   at System.Diagnostics.ProcessThread.get_PriorityLevelCore() in D:\a\1\s\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessThread.Windows.cs:line 75
   at System.Diagnostics.ProcessThread.get_PriorityLevel() in D:\a\1\s\src\System.Diagnostics.Process\src\System\Diagnostics\ProcessThread.cs:line 94
   at System.Diagnostics.Tests.ProcessThreadTests.TestPriorityLevelProperty() in D:\a\1\s\src\System.Diagnostics.Process\tests\ProcessThreadTests.cs:line 177
```

File: src/System.Diagnostics.Process/tests/ProcessThreadTests.Unix.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [Fact]
+        public void TestPriorityLevelProperty_Unix()
+        {
+            CreateDefaultProcess();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => _process.Threads[0].PriorityLevel);
```
Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessThreadTests.Unix.cs]: Why is the getter only tested on OSX and not on Linux?

Comment (stephentoub) [reply src/System.Diagnostics.Process/tests/ProcessThreadTests.Unix.cs]: That's a mistake.  Fixing.

**Pull request merged**